core/src/main/java/org/elasticsearch/search/aggregations/metrics/percentiles/PercentilesAggregationBuilder.java

```diff
@@ -103,7 +104,26 @@
     }
 
     public static AggregationBuilder parse(String aggregationName, XContentParser parser) throws IOException {
-        return PARSER.parse(parser, new PercentilesAggregationBuilder(aggregationName), null);
+        InternalBuilder internal = PARSER.parse(parser, new InternalBuilder(aggregationName), null);
+        // we need to return a PercentilesAggregationBuilder for equality checks to work
+        PercentilesAggregationBuilder returnedAgg = new PercentilesAggregationBuilder(internal.name);
+        setIfNotNull(returnedAgg::valueType, internal.valueType());
+        setIfNotNull(returnedAgg::format, internal.format());
+        setIfNotNull(returnedAgg::missing, internal.missing());
+        setIfNotNull(returnedAgg::field, internal.field());
+        setIfNotNull(returnedAgg::script, internal.script());
+        setIfNotNull(returnedAgg::method, internal.method());
+        setIfNotNull(returnedAgg::percentiles, internal.percentiles());
```

Comment:
Should we also check for empty arrays too?   Or maybe down in the `percentiles()` method?